src/openrct2/ride/RideConstruction.cpp

```diff
@@ -78,6 +78,7 @@ TrackPitch _previousTrackPitchEnd;
 CoordsXYZ _previousTrackPiece;
 
 uint8_t _currentBrakeSpeed;
+uint8_t _currentColourScheme;
```

Comment:
Could you change this into a `RideColourScheme`, and also change TrackElement::GetColourScheme() to return a `RideColourScheme`?

Reply:
I like the idea behind this, but changing the return type of TrackElement::GetColourScheme results in a lot of cascading changes that will need a lot of casts to uint8_t since the value is used as an array index a lot. I will play around with it a bit more, but I think it might wind up being too big a change for this PR.

Reply:
In that case, just make `_currentColourScheme` a `RideColourScheme` and leave `TrackElement::GetColourScheme()` as-is.